### PR TITLE
feat: anchored-region checks local shadow dom for anchor id

### PR DESCRIPTION
**Pull request by scomea**
# Pull Request

## 📖 Description
Anchored region defaulted to checking the light dom to find the anchor element by id.  With this change it will check the local shadow dom for a matching id before checking the light dom.  Matches a behavior we already have in tooltip and should make it easier to use.

### General
- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific
- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts [hunk 1]
```diff
@@ -623,6 +623,12 @@ export class AnchoredRegion extends FoundationElement {
      *  Gets the anchor element by id
      */
     private getAnchor = (): HTMLElement | null => {
+        const rootNode = this.getRootNode();
+
+        if (rootNode instanceof ShadowRoot) {
+            return rootNode.getElementById(this.anchor);
+        }
+
         return document.getElementById(this.anchor);
```
Comment (robarbms) [inline packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:632]: Would this be a problem if the anchored region is inside another component but the anchor isn't. Would writing it like this make it a little more robust?

```suggestion
        return this.getRootNode().getElementById(this.anchor) || document.getElementById(this.anchor);
```

**Review by robarbms — APPROVED** [review]

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts (hunk 1 above)
Comment (scomea) [reply packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:619]: I don't think that makes a difference for nested shadow doms and I'm not sure how to do that and safely make typeScript happy.  I'd need to do something like `(this.getRootNode() as ShadowRoot)` which feels sketch without some type checking.

**Comment by scomea** [comment]
I realized viewport should check shadow dom as well and added that, also fixed some issues with stories.

File: change/@microsoft-fast-foundation-5dacab8f-7ec9-4469-9853-e99b87b243c4.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-5dacab8f-7ec9-4469-9853-e99b87b243c4.json:2]: This and the dependent type need to be “prerelease”

**Review by chrisdholt — APPROVED** [review]
One change needed for the change file.

**Pull request merged**